tests/common/devices/sonic.py

```diff
@@ -639,34 +633,50 @@ def all_critical_process_status(self):
                 all_critical_process[service] = service_critical_process
                 continue
 
-            service_group_process = group_process_results[service]
+            all_critical_process[service] = self.parse_service_status_and_critical_process(
+                service_result=service_results[service],
+                critical_group_list=group_process_results[service]['groups'],
+                critical_process_list=group_process_results[service]['processes']
+            )
 
-            service_result = service_results[service]
-            # If container is not running, stdout_lines is empty
+        return all_critical_process
+
+    def parse_service_status_and_critical_process(self, service_result, critical_group_list,
+                                     critical_process_list):
+        """
+        Parse the result of command "docker exec <container_name> supervisorctl status"
+        and get service container status and critical processes
+        """
+        service_critical_process = {
+            'status': True,
+            'exited_critical_process': [],
+            'running_critical_process': []
+        }
+        # If container is not running, stdout_lines is empty
+        # In this situation, service container status should be false
+        if not service_result['stdout_lines']:
+            service_critical_process['status'] = False
+        for line in service_result['stdout_lines']:
+            pname, status, _ = re.split('\\s+', line, 2)
+            # 1. Check status is valid
+            # Sometimes, stdout_lines may be error messages but not emtpy
             # In this situation, service container status should be false
-            if not service_result['stdout_lines']:
+            # We can check status is valid or not
+            # You can just add valid status str in this tuple if meet later
+            if status not in ('RUNNING', 'EXITED', 'STOPPED', 'FATAL'):
```

Comment:
@lerry-lee If these are some other scenarios we don't consider, could you still add this check here as well? It's for safe check and not harmful. `if pname in critical_group_list or pname in critical_process_list:`

Reply:
@ZhaohuiS Yeah, your concern is right. But this logic is to exclude parsing error messages, if the `'stdout_lines'` is not expected result, then the status is not in valid status tuple, at the same time, the pname is not a valid process name, so it will also skip the judgment logic `if pname in critical_group_list or pname in critical_process_list:` so that the service status can not be changed into False.
At present, our solution is that we can list all valid status in the tuple, currently we found this four status ('RUNNING', 'EXITED', 'STOPPED', 'FATAL'). And I'm not sure what other status are, if you know, you can help to add them.